connect-inject/endpoints_controller_test.go

```diff
@@ -106,6 +107,74 @@ func TestHasBeenInjected(t *testing.T) {
 	}
 }
 
+// TestProcessUpstreamsTLSandACLs enables TLS and ACLS and tests processUpstreams through
+// the only path which sets up and uses a consul client: when proxy defaults need to be read.
+// This test was plucked from the table test TestProcessUpstreams as the rest do not use the client.
+func TestProcessUpstreamsTLSandACLs(t *testing.T) {
+	t.Parallel()
+	nodeName := "test-node"
+
+	masterToken := "b78d37c7-0ca7-5f4d-99ee-6d9975ce4586"
+	caFile, certFile, keyFile := common.GenerateServerCerts(t)
+	// Create test consul server with ACLs and TLS
+	consul, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+		c.ACL.Enabled = true
+		c.ACL.DefaultPolicy = "deny"
+		c.ACL.Tokens.Master = masterToken
+		c.CAFile = caFile
+		c.CertFile = certFile
+		c.KeyFile = keyFile
+		c.NodeName = nodeName
+	})
+	require.NoError(t, err)
+	defer consul.Stop()
+
+	consul.WaitForSerfCheck(t)
+	cfg := &api.Config{
+		Address: consul.HTTPSAddr,
+		Scheme:  "https",
+		TLSConfig: api.TLSConfig{
+			CAFile: caFile,
+		},
+		Token: masterToken,
+	}
+	consulClient, err := api.NewClient(cfg)
+	require.NoError(t, err)
+	addr := strings.Split(consul.HTTPSAddr, ":")
+	consulPort := addr[1]
+
+	ce, _ := api.MakeConfigEntry(api.ProxyDefaults, "pd")
```

Comment:
```suggestion
	ce, _ := api.MakeConfigEntry(api.ProxyDefaults, "global")
```

I don't know how this worked because inside processUpstreams we look for a proxy defaults with name `global`. I think somewhere when you call `ConfigEntries().Set` it's ignoring the `pd` and using `global` but may as well set it to `global` here to avoid confusion.

Reply:
Good catch on this. Not sure how it worked either..